### ISSUE TITLE
What is synse_10_r_unseen_zs.npy and synse_10_r_seen_zs.npy

### ISSUE THREAD

**Issue by WurmD** [skelemoa/synse-zsl#9]
In synse_resources/language_modelling/repo_test_synse_10_r_val/bert/ we have 
`synse_10_r_unseen_zs.npy` - of shape (4821, 10) and  min max [ 0.0077835037 , 0.47233814 ]
and `synse_10_r_seen_zs.npy` - of shape (3240, 10) and min max [ 0.009033936 , 0.7786457 ]

My guess for `synse_10_r_unseen_zs.npy` is the unseen classes probabilities as output of `softmax (SynSEclassifier(sequence_encoder(ShiftGCN_embeddings)))` passed by (is this essentially correct?)
But I have no guess for `synse_10_r_seen_zs.npy`, given it's also of width 10

**Comment by WurmD** [comment]
>  input to the gating model (Section 2.5) is the concatenation of the top k softmax probabilities from the outputs of the seen
and unseen classifiers.  We set k equal to the number of unseen classes and we temperature scale [25] the seen classifier probabilities as well.

So, for the `synse_10_r_seen_zs.npy` you take top 10 ShiftGCN highest classified classes, get the ShiftGCN classification outputs, and temp_scale(..) it, correct?

**Comment by WurmD** [comment]
Then the last question is, of which samples you do this, to train the gating model

**Comment by WurmD** [comment]
>  For training the gating model, we set aside a few samples from the training set and refer to them as the gating train set. Similarly, we set aside a few samples from validation set (gating validation set). We train the gating model using the gating train set and determine the hyperparameters (temperature coefficient, threshold) using the gating validation set.

Set aside? Are these samples used in the training of SynSE?

**Comment by pranaygupta36** [comment]
Yes, you are right about the synse_10_unseen_zs.npy and synse_10_r_seen_zs.npy. 

SynSE is trained twice, 

> 1. First train SYNSE for ZSL in train phase:
> <code> python synse_training.py --ntu 60 --phase train --ss 5 --st r --dataset 'synse_resources/ntu_results/shift_5_r/' --wdir 'synse_resources/language_modelling/synse_5_r' --le bert --ve shift --num_cycles 10 --num_epoch_per_cycle 1700 --latent_size 100 --mode train --gpu 0 </code>
> 2. Then train SYNSE for ZSL in eval phase(Please note the difference in the wdir and dataset values):
> <code> python synse_training.py --ntu 60 --phase val --ss 5 --st r --dataset 'synse_resources/ntu_results/shift_val_5_r/' --wdir 'synse_resources/language_modelling/synse_5_r_val' --le bert --ve shift --num_cycles 10 --num_epoch_per_cycle 1700 --latent_size 100 --mode train --gpu 0 </code>

synse_10_r_seen_zs.npy and synse_10_r_unseen_zs.npy are created in the eval training phase. In the eval phase synse is trained on 100 classes(out of the 110 seen classes) and the remaining 10(from the 110 seen) are treated as unseen. The gating model (effectively an out of distribution detector) is trained on the samples from this 100(seen) + 10(unseen) classes. 

For the final evaluation, synse is retrained on all the 110 seen classes, and the gating model trained before is now used for evaluation of samples from all 120 classes

**Comment by divyanshusharma1709** [comment]
Hi @WurmD! I'll mark this issue as resolved now. Feel free to reopen it if you have further questions!
